arelle/plugin/validate/HMRC/ValidateHmrc.py

```diff
@@ -387,11 +452,14 @@ def _evaluateCode(self, code: str) -> CodeResult:
             result = self._evaluateTextPattern(textValidationPatterns)
         elif code in (CO_AUDIT, LP_AUDIT):
             result = self._evaluateAuditFacts()
-        elif code in (CO_MED_CO, LP_MED_LP):
-            result = self._evaluateMedFact()
+        elif code == CO_DIR_REP:
+            result = self._evaluateDirectorFacts()
+        elif code in (CO_MED_CO, LP_MED_LP, CO_QUAL_AUDIT, LP_QUAL_AUDIT):
+            result = self._evaluateRequiredSingleFact(code)
```

Comment:
Maybe:
```suggestion
        elif code in SINGLE_CONCEPT_EVALUATIONS:
            result = self._evaluateRequiredSingleFact(SINGLE_CONCEPT_EVALUATIONS[code])
```
At least `code in SINGLE_CONCEPT_EVALUATIONS` so we don't have to define the associated codes both here and in `SINGLE_CONCEPT_EVALUATIONS`

Reply:
Pre-existing, but you could update the `TEXT_VALIDATION_PATTERNS` check above with this same pattern.